sdk/identity/Azure.Identity/src/AuthenticationRecord.cs

```diff
@@ -180,6 +189,9 @@ private static async Task<AuthenticationRecord> DeserializeAsync(Stream stream,
                     case ClientIdPropertyName:
                         authProfile.ClientId = prop.Value.GetString();
                         break;
+                    case VersionPropertyName:
+                        authProfile.Version = prop.Value.GetString();
+                        break;
```

Comment:
This should throw an exception in the case that the version is not our only known version.